modules/login/src/main/frontend/src/login/loginMainComponent.js

```diff
@@ -26,63 +26,56 @@ import Logo from "../components/Logo";
+  let [ signInShown, setSignInShow ] = useState(true);
 
-    this.state = {
-      signInShown: true,
-      signUpEnabled : false,
-      isLongForm: !!window.location.pathname.startsWith("/login"),
-      title: document.querySelector('meta[name="title"]').content
-    }
-  }
+  const isLongForm = !!window.location.pathname.startsWith("/login");
+  const title = document.querySelector('meta[name="title"]').content;
 
   // Toggle between sign in and sign up
-  handleSwap = () => {
-    this.setState(prevState => ({
-      signInShown: !prevState.signInShown,
-    }));
+  let handleSwap = () => {
+    setSignInShow(!signInShown);
   }
 
-  render () {
-    const { classes, selfContained } = this.props;
-
-    return (
-        <Paper className={`${classes.paper}  ${selfContained ? classes.selfContained : ''}`} elevation={0}>
-          <Grid container direction="column" spacing={3} alignItems="center" alignContent="center">
-            <Logo maxWidth="200px" component={Grid}/>
-            <Grid>
-            { this.state.signInShown ? <SignIn handleLogin={this.props.handleLogin} redirectOnLogin={this.props.redirectOnLogin}/> : <SignUpForm loginOnSuccess={true} handleLogin={this.props.handleLogin} /> }
-            </Grid>
-            { this.state.isLongForm && (!this.state.signInShown || this.state.signUpEnabled) &&
-              <Grid>
-                <Button
-                  variant="outlined"
-                  fullWidth
-                  className={classes.main}
-                  onClick={this.handleSwap}
-                 >
-                  { this.state.signInShown ?  "Sign up" : "Sign In" }
-                </Button>
-              </Grid>
-            }
-            { this.state.isLongForm &&
-            <Grid>
-              <Breadcrumbs separator="by" className={classes.appInfo}>
-                <Typography variant="subtitle2">{this.state.title}</Typography>
-                <Tooltip title="DATA Team @ UHN">
-                  <a href="https://uhndata.io/" target="_blank">
-                    <img src="/libs/cards/resources/media/default/data-logo_light_bg.png" width="80" alt="DATA" />
-                  </a>
-                </Tooltip>
-              </Breadcrumbs>
-            </Grid>
-            }
+  return (
+    <Paper className={`${classes.paper}  ${selfContained ? classes.selfContained : ''}`} elevation={0}>
+      <Grid container direction="column" spacing={3} alignItems="center" alignContent="center">
+        <Logo maxWidth="200px" component={Grid}/>
+        <Grid>
+        { signInShown ?
+          <SignIn handleLogin={handleLogin} redirectOnLogin={redirectOnLogin}/>
+          :
+          <SignUpForm loginOnSuccess={true} handleLogin={handleLogin} />
+        }
+        </Grid>
+        { isLongForm && !signInShown &&
+          <Grid>
+            <Button
+              variant="outlined"
+              fullWidth
+              className={classes.main}
+              onClick={handleSwap}
```

Comment:
I would get rid of `handleSwap` as it's only used in this one place.
```suggestion
              onClick={() => setSignInShow(!signInShown)}
```